src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs

```diff
@@ -80,6 +80,14 @@ private void OnChanged(object sender, FileSystemEventArgs e)
 
         private void OnFileSystemEntryChange(string fullPath)
         {
+            var fileSystemInfo = new FileInfo(fullPath);
+            if (fileSystemInfo.Name.StartsWith(".") ||
+                (fileSystemInfo.Exists && fileSystemInfo.Attributes.HasFlag(FileAttributes.Hidden)))
```

Comment:
This looks like it should go in some static helper to re-use in the several places it's used.

Also, I think this needs to check for `FileAttributes.System`, which is a "super hidden" file.